src/lsp/cobol_cfg/cfg_builder.ml

```diff
@@ -0,0 +1,542 @@
+(******************************************************************************)
+(*                                                                            *)
+(*     Copyright (c) 2021-2023 OCamlPro SAS                                   *)
+(*                                                                            *)
+(*     All rights reserved.                                                   *)
+(*     This file is distributed under the terms of the                        *)
+(*     OCAMLPRO-NON-COMMERCIAL license.                                       *)
+(*                                                                            *)
+(******************************************************************************)
+
+open Cobol_unit
+open Cobol_common.Srcloc.INFIX
+open Cobol_common.Srcloc.TYPES
+open Cobol_unit.Types
+open Cfg_jumps
+module NEL = Cobol_common.Basics.NEL
+
+(* TYPES AND HELPERS *)
+
+type qualname = Cobol_ptree.qualname
+
+type display_name_type =
+  | Full
+  | Short
+
+let qn_to_strings : qualname -> string * string = function
+  | Name { payload; _ } -> payload, ""
+  | Qual({ payload = para; _ }, Name { payload = sec; _ }) -> para, sec
+  | _ -> raise @@ Invalid_argument
+      "qn with more than 2 qualification levels cannot \
+       come from a paragraph or section"
+
+let qn_to_fullname qn =
+  let name, qual = qn_to_strings qn in
+  if qual == ""
+  then name
+  else name ^ " IN " ^ qual
+
+let incr ref =
```

Comment:
This redefinition is to be avoided. Maybe this could be named `next_int` or something like this.

Could be even better to use [`Seq.ints 0`](https://ocaml.org/manual/5.2/api/Seq.html#1_Sequencesofintegers)